docs/Lexicon.rst

```diff
@@ -144,6 +144,14 @@ source code, tests, and commit messages. See also the `LLVM lexicon`_.
     themselves. They can be compared across declarations but cannot be used
     directly from within the context.
 
+  irrefutable pattern
+    A pattern that always matches. These patterns either bind to a variable or
+    perform structural modification, e.x.:
+
+    1. ``case _:``.
+    2. ``case let x:``.
+    3. ``case (_, _):``.
```

Comment:
The last example can fail if the input is `Any`, but this wasn't my understanding of refutable vs. irrefutable patterns anyway. I thought it was entirely contextual—`let` patterns vs. `case` patterns.